### PR TITLE
Adapt softmin layer

### PR DESCRIPTION
**Pull request by aadi-raj**
An attempt to adapt softmin layer according to changes in #2777.

### CODE REVIEW

File: src/mlpack/methods/ann/layer/serialization.hpp [hunk 1]
```diff
@@ -67,6 +67,7 @@
     CEREAL_REGISTER_TYPE(mlpack::PaddingType<__VA_ARGS__>); \
     CEREAL_REGISTER_TYPE(mlpack::RBFType<__VA_ARGS__>); \
     CEREAL_REGISTER_TYPE(mlpack::SoftmaxType<__VA_ARGS__>); \
+    CEREAL_REGISTER_TYPE(mlpack::SoftminType<__VA_ARGS__>); \
```
Comment (rcurtin) [inline src/mlpack/methods/ann/layer/serialization.hpp:70]: @AdarshSantoria did we forget to do this for the PReLU layer?  If so, want to open a simple PR to fix it?

**Review by rcurtin — APPROVED** [review]
Very nice, thanks @aadi-raj. I think everything is good here. Do you want to add a note to `HISTORY.md`?

File: src/mlpack/methods/ann/layer/serialization.hpp (hunk 1 above)
Comment (AdarshSantoria) [reply src/mlpack/methods/ann/layer/serialization.hpp:70]: No, I had added this for PReLU in #3420.

Comment (rcurtin) [reply src/mlpack/methods/ann/layer/serialization.hpp:70]: Weird, the diff here doesn't show it. Anyway, you are right, it is added correctly to `master`. Thanks :)

**Pull request merged**